### ISSUE TITLE
First argument of mix() is not optional

### ISSUE THREAD

**Issue by jozsefs** [styled-components/polished#244]
- `polished` version: 1.7.0

Mixin/Helper/Shortand you were using and how you were using it:

```javascript
polished.mix('#fff', '#f00');
```

What you are seeing:

```javascript
polished.js:1480 Uncaught Error: Passed an incorrect argument to a color function, please pass a string representation of a color.
    at parseToRgb (polished.js:1480)
    at mix (polished.js:2213)
    at fn (polished.js:1874)
    at <anonymous>:1:1
```

What you expected to see:

```javascript
"#ff7f7f"
```

The first argument of the `mix()` function (the `weight` argument) is not "optional" however in the docs it's explicit that it should be optional and it's implied from the code.
It is optional in the sense that if you pass `undefined` then the default `0.5` will be used but you can't use it with 2 arguments only. 

A solution could be to either refactor the function to check the number of arguments or to update the docs + remove the question mark from the related code. Also the 3rd option is to just modify the docs to reflect that you need to pass `undefined` which doesn't make much sense to me. Neither of these options should be a breaking change.

**Comment by morajabi** [comment]
@mxstbr Sorry, how can we have multiple definitions for a single function in Flow?

**Comment by bhough** [comment]
@jozsefs This is a slightly annoying side effect of making the color methods composable. Since the color getting passed in from another color method is passed as the last argument when composing, we have to put these values first. Let me think a bit on the best way to handle this.

**Comment by morajabi** [comment]
@bhough Isn't it better to have another issue/discussion specifically on API design topic?

**Comment by bhough** [comment]
@morajabi we won’t change the API for these, it is too ingrained in how they all work. This is specifically how to handle the odd case of `mix` since it is the only one off the top of my head that is trying to set a default for the first argument.

**Comment by bhough** [comment]
@Prinzhorn Already ahead of you! The docs will be updated soon: https://github.com/styled-components/polished/blob/docs-types-review/src/color/mix.js

**Comment by Prinzhorn** [comment]
@bhough nice 😘

**Comment by bhough** [comment]
Addressed by #327 and merged into `version-2`
